kernel/syscall-defs.h

```diff
@@ -3,6 +3,23 @@
 
 typedef unsigned long long syscall_arg;
 
+typedef struct {
+  unsigned long addr;
+  unsigned long r1;
+  unsigned long r2;
+} ipc_msg;
+
+typedef struct {
+  ipc_msg send; /* Filled in by the user */
+  ipc_msg recv; /* Filled in by the kernel on a receive */
+} sendrecv_op;
+
+typedef enum {
+  SEND_FAILED = 0,
```

Comment:
Can you make 0 the successful case, so 'if (sendrecv())' works?

Reply:
Sure